WordPress/gutenberg
# Fix: Correct border styling for RTL environments

## What?
The PR fixes the inversion of border styles in the Button block for RTL languages in the WordPress block editor.

Related to #64512

## Why?
This PR is necessary to address a critical bug in the WordPress block editor that affects RTL (Right-to-Left) languages, such as Farsi. The issue causes border styles in the Button block to be applied inversely, leading to incorrect visual representation and user confusion. Fixing this bug ensures that the border settings are accurately applied in RTL environments, improving the overall user experience and visual consistency.

## How?
The PR addresses the issue by correcting the logic that applies border styles to the Button block in RTL environments. Specifically, the implementation changes the way borders are assigned in the CSS, ensuring that left and right borders are applied correctly according to the RTL context.

In the previous code, there was a conditional check that incorrectly swapped the border-left and border-right properties based on the RTL setting. The updated implementation removes this conditional inversion, directly assigning the correct border styles to the corresponding sides (border-left and border-right), thereby fixing the visual representation of borders in RTL languages.

## Testing Instructions
1. Open a post or page in the WordPress editor.
2. Ensure your WordPress site is set to an RTL language (e.g., Farsi or Arabic) via `Settings` > `General` > "Site Language."
3. Insert a Button block into the post or page.
4. In the Button block settings, apply custom border sizes to the left, right, top, and bottom borders.
5. Save the post or page.
6. Preview the post or page on the frontend and ensure that the border styles are correctly applied: 
   - The left border appears on the left side.
   - The right border appears on the right side.
7. Confirm that the visual representation matches the settings specified in the block editor.
8. Optionally, switch to an LTR language (e.g., English) and repeat steps 3-7 to compare the behavior.

### Testing Instructions for Keyboard


## Screenshots or screencast 

### Before:

![before](https://github.com/user-attachments/assets/6e335623-d601-4589-987a-3c98cf4a8d09)

### After
![after](https://github.com/user-attachments/assets/f69d0c65-f1ab-4cbe-a28b-6d825d47b595)
